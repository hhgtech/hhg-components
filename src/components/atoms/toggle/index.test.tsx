import React from 'react'
import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import MockTheme from './../../../mockTheme'

import { Toggle } from '.'

afterEach(cleanup)

test('it renders', () => {
  const toggle = render(
    <MockTheme>
      <Toggle
        id="test-toggle"
        checked
        onChange={() => console.log('onChange')}
      />
    </MockTheme>,
  )

  const self = toggle.container.firstChild

  expect(self).toMatchSnapshot()

  expect(self.firstChild).toHaveAttribute('checked')
})
