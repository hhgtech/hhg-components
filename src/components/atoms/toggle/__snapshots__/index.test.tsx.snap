// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 input[type='checkbox'] {
  width: 0;
  height: 0;
  visibility: hidden;
}

.c0 label {
  position: relative;
  display: block;
  width: 40px;
  height: 24px;
  background: #c3cbd4;
  border-radius: 100px;
  cursor: pointer;
  text-indent: -9999px;
}

.c0 label:after {
  position: absolute;
  top: 2px;
  left: 2px;
  width: 20px;
  height: 20px;
  background: #fff;
  border-radius: 90px;
  content: '';
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c0 input:checked + label {
  background: #2d87f3;
}

.c0 input:checked + label:after {
  left: calc(100% - 2px);
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

<div
  class="c0"
>
  <input
    checked=""
    id="test-toggle"
    type="checkbox"
  />
  <label
    for="test-toggle"
  >
    Toggle
  </label>
</div>
`;
